include/benchmark/benchmark_api.h

```diff
@@ -230,7 +230,7 @@ BENCHMARK_UNUSED static int stream_init_anchor = InitializeStreams();
 // expression from being optimized away by the compiler. This function is
 // intended to add little to no overhead.
 // See: https://youtu.be/nXaxk27zwlk?t=2441
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__pnacl__) && !defined(EMSCRIPTEN)
```

Comment:
OK I probably overstated "not supporting inline assembly" as a bug. But if Clang parses "asm volatile(...)" it should probably also support generating code, at least for empty instructions.

I just tried compiling an file which used inline assembly with `clang++ -target x86_64-unknown-nacl` and it accepted it. I can't test linking and running it for obvious reasons, but I'm still skeptical of this change.